### PR TITLE
[helm_lib] Properly update CSI controller Deployment

### PR DESCRIPTION
**Pull request by zuzzas**
## Description

Update CSI controller without creating a new one.

## Why do we need it, and what problem does it solve?

Since controller is not a StatefulSet anymore, there can be unfortunate complications while trying to update it. Such as:
```
2022/04/27 12:39:33 failed to listen: listen tcp 192.168.176.29:12302: bind: address already in use
```

## Changelog entries

```changes
section: helm_lib
type: fix
summary: Update CSI controller without creating a new one.
impact_level: low
```

### CODE REVIEW

**Review by kvaps — COMMENTED** [review]
Wouldn't it be better just to remove `hostNetwork: true`?

**Comment by zuzzas** [comment]
Definitely not. Such an important component should rely on network availability.

**Comment by kvaps** [comment]
> Definitely not. Such an important component should not rely on network availability.

But it is still rely on both Kubernetes services and and pod networking, eg. the same kube-dns is running with the podNetwork.

From my experience almost all CSI-controllers are use no host networking.

**Comment by zuzzas** [comment]
We deploy all crucial component in the `hostNetwork: true` mode. Such a discussion and discussion potential ramifications should be held in a separate issue.

File: helm_lib/templates/_csi_controller.tpl [hunk 1]
```diff
@@ -91,6 +91,10 @@ spec:
   selector:
     matchLabels:
       app: {{ $fullname }}
+  strategy:
```
[the remark this answers is not in the document]
Comment (zuzzas) [reply helm_lib/templates/_csi_controller.tpl:94]: Why indeed...

Fixed!

**Comment by kvaps** [comment]
OK, got it!

**Review by konstantin-axenov — APPROVED** [review]

**Comment by z9r5** [comment]
Cherry-picked into release-1.32

**Pull request merged**